### PR TITLE
Backup-pistol overhaul, now with .380!

### PR DESCRIPTION
**Pull request by tenmillimaster**
#### Summary

Example: ```SUMMARY: Content "Adds .380 ACP ammo, a number of pocket pistols and corrects existing ones."```
#29434 
#### Purpose of change

While trawling through various sources, I noticed that .380 is still a significant caliber in the new england area, at least according to ATF crime statistics for seized weapons. As such, this PR seeks to add .380 ACP and give some attention to the tiny guns its commonly used in,  so called 'pocket pistols'.  In addition, since .380 ACP is known to be able to be fired in 9mm and 9mm makarov guns generally without incident (but usually wholly unrecommended), I've included that ~~feature~~ IRL bug here, likely as the most common usage the given survivor would get out of this ammo. CDDA is a game about desperation and making do, after all, and worse mismatches exist currently in game.

#### Describe the solution
Where appropriate, volumes were calculated by approximating a 'box' that a given pistol would fill. This is still overestimating slightly, but much better than defaulting to the lowest unit available. 

I've also added a new template for such pistols, 'pistol_backup', which removes many of the mod slots.

#### Additional context

I swear I'm not a keltec fanboy- they've just been in the business for the longest it seems, capitalizing on the wide legalization of concealed carry at the right time and offering affordable pistols in a niche major manufacturers were ignoring.

EDIT: Here's the firearms trace data for the New England states, showing that .380 is among the top traced calibers.  
<https://www.atf.gov/docs/undefined/mawebsite17183904pdf/download>
<https://www.atf.gov/docs/undefined/ctwebsite17183889pdf/download>
<https://www.atf.gov/file/130191/download>
<https://www.atf.gov/docs/undefined/mewebsite17183891pdf/download>
<https://www.atf.gov/docs/undefined/riwebsite17183909pdf/download>
<https://www.atf.gov/resource-center/docs/undefined/vtwebsite17183910pdf/download>

TODO:
Finish up magazines
Unloading and reloading recipes
See if deleting the defaulting to torso coverage line in item.cpp and adding armor data to the shoulder strap fixes my belt clip, it keeps covering legs and torsos
Stick the guns in the item groups

### CODE REVIEW

File: data/json/items/gunmod/accessories.json [hunk 1]
```diff
@@ -68,5 +68,22 @@
...
+    "weight": 40,
+    "volume": 1,
+    "integral_volume": 0,
+    "price": 2000,
+    "material": "steel",
+    "symbol": ":",
+    "color": "light_gray",
+    "armor_data": { "covers": [ "LEG_EITHER" ] },
+    "location": "accessories",
+    "mod_targets": [ "rugerlcp", "kp32", "kp3at", "kpf9", "cop_38", "moss_brownie", "moss_brownie" ],
```
Comment (Rail-Runner) [inline data/json/items/gunmod/accessories.json]: You specified "moss_brownie" twice here, probably not necessary.

**Comment by Throwaway-name** [comment]
The pocket pistol additions should be noted in your summary line.

File: data/json/items/gunmod/accessories.json (hunk 1 above)
Comment (tenmillimaster) [reply data/json/items/gunmod/accessories.json]: Thanks, missed that.

File: data/json/items/gun/380.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    "weight": 240,
+    "volume": "231ml",
+    "price": 25000,
+    "ammo": "380"
+  },
+  {
+    "id": "fn1910",
+    "copy-from": "pistol_backup",
+    "type": "GUN",
+    "name": "FN 1910 .380",
```
Comment (Night-Pryanik) [inline data/json/items/gun/380.json]: This little guy should be a VERY rare find considering its production was stopped at 1983.

Comment (tenmillimaster) [reply data/json/items/gun/380.json]: @Night-Pryanik absolutely, that's the idea. Mansions only, basically, and with a tiny item group chance too.

Comment (Night-Pryanik) [reply data/json/items/gun/380.json]: I think one variant is enough. Choose the one you feel more fit.

**Comment by tenmillimaster** [comment]
Added the source data I'm using to justfiy .380's relevance in the area, from the ATF trace data  in 2017.

File: data/json/itemgroups/guns.json [hunk 3]
```diff
@@ -14,7 +14,11 @@
       [ "sw_22", 15 ],
       [ "sw_610", 20 ],
       [ "sw_619", 25 ],
-      [ "p226_357sig", 20 ]
+      [ "p226_357sig", 20 ],
+      [ "kp32", 10 ],
+      [ "kp3at", 35 ],
+      [ "rugerlcp", 35 ],
+      [ "kpf9", 35 ],
     ]
```
Comment (Throwaway-name) [inline data/json/itemgroups/guns.json]: Bad commas.

File: data/json/itemgroups/guns.json [hunk 4]
```diff
@@ -363,7 +368,11 @@
       [ "remington_870", 20 ],
       [ "sig_40", 20 ],
       [ "usp_45", 10 ],
-      [ "usp_9mm", 10 ]
+      [ "usp_9mm", 10 ],
+      [ "kp32", 1 ],
+      [ "kp3at", 2 ],
+      [ "rugerlcp", 2 ],
+      [ "kpf9", 3 ],
     ]
```
Comment (Throwaway-name) [inline data/json/itemgroups/guns.json]: Bad commas.

File: data/json/itemgroups/guns.json (hunk 3 above)
Comment (tenmillimaster) [reply data/json/itemgroups/guns.json]: TY.

**Comment by tenmillimaster** [comment]
I think this is a wrapped up now after having played with it a little. I've added a very, VERY cheap ~~hand grenade~~ saturday night special, and a belly band holster. The Belt clip isn't functioning as intended still- it mechanically functions as a sling since I can't get the hard code stuff sorted out, but it still works as a low encumbrance carry option, albeit in the most oft used slot. I'll tackle that in the future.

**Comment by tenmillimaster** [comment]
I've added the MAC-10/11's iconic folding wire stock. This should probably be added as a recipe, and as an installed gun mod for a number of guns (Uzi, AK's with underfolder stocks), maybe some equivalents of the B+T USW for the machine pistol (Leadworks machinepistols). In another PR.

**Pull request merged**